### PR TITLE
First draft of management cluster getting started guide

### PR DESCRIPTION
**Pull request by chrisnegus**
Documentation for [Add support to create and mange a management cluster](https://github.com/aws/eks-anywhere/issues/94).

Still not decided if it will replace or coexist with initial getting started guide.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/573#" title="Author self-approved">chrisnegus</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[docs/OWNERS](https://github.com/aws/eks-anywhere/blob/main/docs/OWNERS)~~ [chrisnegus]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:22]: You may want to add " and reduces provider permissions for workload clusters."

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:38]: sigb? Is that a special name?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:45]: I thought our default was 3 etcd, 2 control plane, and 3 worker. Did that change?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:52]: This is a really big example. Is there a way we can trim it down in the doc to just the fields someone will need to modify? Or maybe a table that shows the values in a table?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:219]: Worth adding a note that etcd nodes don't have a version because they don't run the kublet?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:239]: Do we not have a CLI command to generate a workload cluster config? That would be really helpful.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:15]: I believe you need to surround the alert in `{{...}}`
```suggestion
{{% alert title="Important" color="primary" %}}
```

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:24]:
```suggestion
{{% /alert %}}
```

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:49]:
```suggestion
   {{% alert title="Note" color="primary" %}}
   The config file shown below also notes which values should be changed if you want to reuse the file to create workload clusters.
   {{% /alert %}}
```

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:205]: nit: blank line, which will show up in the docs as it's enclosed in back ticks

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:64]: maybe @mdsgabriel can answer this, but why do these have to be different? These are used for internal cluster networking and should conflict with each other right?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:199]: nit: do we really need this? It doesn't show anything specific about the management cluster

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (abhinavmpandey08) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:220]: nit:
```suggestion
   eksa-system sigb-b2xyz          vsphere://xxxxx    Running  v1.21.2-eks-1-21-5
   eksa-system sigb-etcd-r9b42     vsphere://xxxxx    Running  
   eksa-system sigb-md-8-6xr-rnr   vsphere://xxxxx    Running  v1.21.2-eks-1-21-5
```

**Review by abhinavmpandey08 — COMMENTED** [review]
Will this eventually replace this page? https://anywhere.eks.amazonaws.com/docs/getting-started/production-environment/

Also there are a lot of references to `sigb` which we probably need to generalize

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:38]: can change this to cluster-name for example

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:170]: maybe user@xyz.ant.example.com

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:38]: Not a special name. I was just using the name that came in the yaml file I got from Gabriel. It is "prod" in the other getting started example. I can change it throughout to "prod" or something else.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:22]: Good point. Done.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:43]: The text says 3 cp and 3 worker nodes. The example shows 2 cp, 2 worker nodes, and 3 etcd. Is there a reason for only two CP nodes? Do you think  3 etcd, 2 control plane, and 3 worker is best for the default?

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:50]: I cut out the gitops content, which makes it somewhat shorter. The reason I don't want to put it in a table is that I think it makes more sense visually this way. The reader would have to figure out which `cidrBlocks` and `name` values I'm talking about. I'll look for ways to trim it, but I think this is useful.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:180]: True. I'll cut it.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:186]: Good catch. Fixed.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:220]: My understanding is that it is not much different than the management cluster yaml file. But I'll ask @mdsgabriel about it.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:43]: CP doesn't need to reach quorum so there isn't a requirement on an odd number. They just need to be HA.

I will leave the default decision up to someone else on the team but I _thought_ 3:2:3 were the defaults for vSphere

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (rothgar) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:50]: I agree it's useful because a lot of the key names are repeated. I hope the comments are visually noticeable enough and people don't try to copy/paste the example

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:41]: users can also generate new config using 

```
eksctl generate clusterconfig <cluster name> -p vsphere
```

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:59]: pods.cirdBlocks doesn't have to be unique

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:62]: services.cidrBlocks doesn't have to be unique

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:215]: > Do we not have a CLI command to generate a workload cluster config? That would be really helpful.
no. the same spec can be used to crate a management cluster or a workload cluster.

if the optional flag --kubeconfig is passed to the create command then the cli will create a workload cluster

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (mdsgabriel) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:217]: it is faster if users call the generate command passing a different name

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:43]: I'll put 3:2:3 then.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: I'm going to cut this whole example and point readers to the vsphere configuration page instead. Then I'll tell them how many nodes minimum the should use and what the optional config settings are.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: I think I have it now.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:148]: I'll make a note to myself to add this information to a troubleshooting section.

**Review by chrisnegus — COMMENTED** [review]
I think all the review comments are in. We need to decide if this will replace the getting started guide or be parallel to it.

**Comment by abhay-krishna** [comment]
@chrisnegus: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-docs-presubmit | 9a3f141cad512e791d7cea69708ffa888dd16b33 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/573/eks-anywhere-docs-presubmit/1457829310311174144) | `/test eks-anywhere-docs-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=573). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:chrisnegus).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (vivek-koppuru) [inline docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md:18]:
```suggestion
Separating management features into a separate, persistent management cluster
```

**Comment by abhay-krishna** [comment]
@chrisnegus: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-docs-presubmit | 19772f044e080c9218488ba60031e42ca2ff18e4 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/573/eks-anywhere-docs-presubmit/1458163577775984640) | `/test eks-anywhere-docs-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=573). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:chrisnegus).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@chrisnegus: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-docs-presubmit | 57af64c57b79388dba0b0f37c415eb986e5809fc | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/573/eks-anywhere-docs-presubmit/1458170809951457280) | `/test eks-anywhere-docs-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=573). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:chrisnegus).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: Fixed

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: Fixed

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: Fixed.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: I removed the example.

**Comment by abhay-krishna** [comment]
@chrisnegus: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-docs-presubmit | b94a8d238badf01a6f8157b4043441d3743531b1 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/573/eks-anywhere-docs-presubmit/1458172762454495232) | `/test eks-anywhere-docs-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=573). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:chrisnegus).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: Done.

File: docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md
Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/gs-mgmt-cluster.md]: Fixed.

File: docs/content/en/docs/getting-started/production-environment/_index.md
Comment (rothgar) [inline docs/content/en/docs/getting-started/production-environment/_index.md:13]:
```suggestion
* Keeps the management cluster in place so you can use it later to modify, upgrade, and delete workload clusters
```

Comment (chrisnegus) [reply docs/content/en/docs/getting-started/production-environment/_index.md:13]: Done.

**Review by chrisnegus — COMMENTED** [review]
@mdsgabriel Please check and lgtm if you think it is ready to merge.

**Review by mdsgabriel — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/573#" title="Author self-approved">chrisnegus</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/573#pullrequestreview-802745709" title="Approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[docs/OWNERS](https://github.com/aws/eks-anywhere/blob/main/docs/OWNERS)~~ [chrisnegus,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
